src/v/storage/disk_log_impl.cc

```diff
@@ -278,9 +278,6 @@ ss::future<> disk_log_impl::do_compact(compaction_config cfg) {
             continue;
         }
 
-        co_await _stm_manager->ensure_snapshot_exists(
-          seg->offsets().committed_offset);
-
```

Comment:
I wanted to minimize the upcalls from the storage layer to the upper transactional layer to minimize a risk of a deadlock which we saw with one of the customers. I pretty sure the deadlock was fixed with https://github.com/vectorizedio/redpanda/pull/1954/commits/cd67056720db45ca6d874ee1e9f455d033c09278 and https://github.com/vectorizedio/redpanda/pull/1954/commits/04fd39f4eeb95170d3eaf1b00edac8106481a827 but I want to play safe and since we block transaction functionality for the compacted topics it's safer to not to do work the client can't benefit from.

Reply:
just to be clear, the action item for this is to reflect the intention of this removal in the commit message. looking at the patch and looking at the commit message, this two line removal looks out of place.